### PR TITLE
Added Template property to RouteTemplate

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Related to #211 
@rynowak

### CODE REVIEW

File: src/Microsoft.AspNet.Routing/Template/RouteTemplate.cs [hunk 1]
```diff
@@ -13,13 +13,15 @@ public class RouteTemplate
     {
         private const string SeparatorString = "/";
 
-        public RouteTemplate(List<TemplateSegment> segments)
+        public RouteTemplate(List<TemplateSegment> segments, string template)
```
Comment (rynowak) [inline src/Microsoft.AspNet.Routing/Template/RouteTemplate.cs]: pass the template first :+1:

File: src/Microsoft.AspNet.Routing/Template/RouteTemplate.cs [hunk 2]
```diff
@@ -37,6 +39,8 @@ public RouteTemplate(List<TemplateSegment> segments)
             }
         }
 
+        public string Template { get; private set; }
+
         public IList<TemplatePart> Parameters { get; private set; }
 
         public IList<TemplateSegment> Segments { get; private set; }
```
Comment (rynowak) [inline src/Microsoft.AspNet.Routing/Template/RouteTemplate.cs]: Change all of these to `{ get; }` #PunishmentForTouchingThisCode

File: test/Microsoft.AspNet.Routing.Tests/Template/TemplateParserTests.cs [hunk 3]
```diff
@@ -21,7 +21,7 @@ public void Parse_SingleLiteral()
             // Arrange
             var template = "cool";
 
-            var expected = new RouteTemplate(new List<TemplateSegment>());
+            var expected = new RouteTemplate(new List<TemplateSegment>(), template);
```
Comment (rynowak) [inline test/Microsoft.AspNet.Routing.Tests/Template/TemplateParserTests.cs]: Is there an equality comparer somewhere? Update that too.

**Comment by rynowak** [comment]
:watch:

**Comment by ajaybhargavb** [comment]
:up: :date:

**Comment by rynowak** [comment]
:shipit:

**Pull request merged**